api/common/v1alpha1/ref.go

```diff
@@ -43,9 +43,17 @@ func (k TargetRefKind) Less(o TargetRefKind) bool {
 }
 
 func AllTargetRefKinds() []TargetRefKind {
-	return maps.Keys(order)
+	keys := maps.Keys(order)
+	sort.Sort(TargetRefKindSlice(keys))
+	return keys
 }
 
+type TargetRefKindSlice []TargetRefKind
```

Comment:
This is an enhancement to a previous PR at https://github.com/kumahq/kuma/pull/9687